interface/forms/fee_sheet/new.php

```diff
@@ -1359,49 +1254,81 @@ function setJustify(seljust) {
+<?php } else { // unbilled with no charges ?>
+<input type='submit' name='bn_save_close' value='<?php echo xla('Save and Close'); ?>' />
+<?php } // end no charges ?>
 &nbsp;
-<?php if (!$hasCharges) { ?>
-<input type='submit' name='bn_save_close' value='<?php echo xla('Mark as Billed');?>' />
+<?php } // end ippf-only ?>
+<input type='submit' name='bn_refresh' onclick='return this.clicked = true;' 
+value='<?php echo xla('Refresh');?>'>
 &nbsp;
-<?php } ?>
-<input type='submit' name='bn_refresh' value='<?php echo xla('Refresh');?>'>
+<?php } else { // visit is billed ?>
+<?php if ($fs->hasCharges) { // billed with charges ?>
+<input type='button' value='<?php echo xla('Show Receipt'); ?>'
+ onclick="top.restoreSession();location='../../patient_file/pos_checkout.php?framed=1<?php
+ echo "&ptid=" . urlencode($fs->pid) . "&enc=" . urlencode($fs->encounter); ?>'" />
 &nbsp;
-<?php } ?>
-
+<input type='submit' name='bn_reopen' onclick='return this.clicked = 2;'
+ value='<?php echo xla('Void Checkout and Re-Open'); ?>' />
+&nbsp;
+<?php } else { ?>
+<input type='submit' name='bn_reopen' onclick='return this.clicked = true;'
+ value='<?php echo xla('Re-Open Visit'); ?>' />
+&nbsp;
+<?php } // end billed without charges ?>
+<input type='submit' name='bn_addmore' onclick='return this.clicked = true;'
+ value='<?php echo xla('Add More Items'); ?>' />
+&nbsp;
+<?php } // end billed ?>
+<input type='hidden' name='form_has_charges' value='<?php echo $fs->hasCharges ? 1 : 0; ?>' />
 <input type='hidden' name='form_checksum' value='<?php echo $current_checksum; ?>' />
 <input type='hidden' name='form_alertmsg' value='<?php echo attr($alertmsg); ?>' />
 
 <input type='button' value='<?php echo xla('Cancel');?>'
- onclick="top.restoreSession();location='<?php echo "$rootdir/patient_file/encounter/$returnurl" ?>'" />
-
-<?php if ($code_types['UCSMC']) { ?>
-<p style='font-family:sans-serif;font-size:8pt;color:#666666;'>
-&nbsp;<br>
-<?php echo xlt('UCSMC codes provided by the University of Calgary Sports Medicine Centre');?>
-</p>
-<?php } ?>
+ onclick="top.restoreSession();location='<?php echo $GLOBALS['form_exit_url']; ?>'" />
 
 </center>
 
 </form>
+
+<?php if (empty($_POST['running_as_ajax'])) { ?>
+
 <script language='JavaScript'>
+setSaveAndClose();
+
 <?php
 echo $justinit;
 if ($alertmsg) {
   echo "alert('" . addslashes($alertmsg) . "');\n";
 }
 ?>
 </script>
+
+<?php } // end not ajax ?>
+
 </body>
 </html>
+<?php if (!empty($_POST['running_as_ajax'])) exit; ?>
 <?php require_once("review/initialize_review.php"); ?>
+<?php require_once("code_choice/initialize_code_choice.php"); ?>
+<?php if ($GLOBALS['ippf_specific']) require_once("contraception_products/initialize_contraception_products.php"); ?>
+<script>
+var translated_price_header="<?php echo xlt("Price");?>";
+</script>
+<!-- Sorry I just can't handle this. -Rod -->
+<!-- script type="text/javascript" src="<?php echo $web_root;?>/interface/forms/fee_sheet/js/layout_changes.js"> /script -->
```

Comment:
If not using, then rec removing them. Can always place a comment on the commit comments.